samza-core/src/main/java/org/apache/samza/storage/StoreBackupEngine.java

```diff
@@ -0,0 +1,185 @@
+package org.apache.samza.storage;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.metrics.MetricsRegistryMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.conf.*;
+
+// Class to hold common methods and variables for backup storage and restore process
+public class StoreBackupEngine {
+  protected static final Logger LOGGER = LoggerFactory.getLogger(StoreBackupEngine.class);
+  protected FileSystem fs;
+  protected Path hdfsPath;
+  protected Path dbPath;
+  public StoreBackupMetrics metrics;
+
+  public StoreBackupEngine(Config storeConfig) {
+    metrics = new StoreBackupMetrics("backup", new MetricsRegistryMap("registrymap"));
+
+    // Setup fs configuration settings
+    Configuration conf = new Configuration();
+    // If core site path file is specified
+    if (storeConfig.containsKey("hdfs.coreSitePath")){
+      Path coreSitePath = new Path(storeConfig.get("hdfs.coreSitePath"));
+      conf.addResource(coreSitePath);
+    } else {
+      // Otherwise, set a default localhost port for fs.defaultFS - hdfs://localhost:9000
+      conf.set("fs.default.name", "hdfs://localhost:9000");
+    }
+
+    // Initialize path variables
+    // Method: config.get(key, defaultVal)
+    dbPath = new Path(storeConfig.get("rocksdb.dbDir", System.getProperty("java.io.tmpdir") + "/DB"));
```

Comment:
We shouldn't add this config. This is currently set based on some configs and environment variables, search SamzaContainer for getLoggedStorageBaseDir to see how.